src/actions/addCartList.ts

```diff
@@ -0,0 +1,62 @@
+"use server";
+
+import { revalidatePath } from "next/cache";
+import { cookies } from "next/headers";
+
+import { createServerActionClient } from "@supabase/auth-helpers-nextjs";
+
+import { prisma } from "../lib/prisma";
+import { Database } from "../types/SupabaseTypes";
+
+const createCartListItem = async (cartListId: number, ingredientIds: number[]) => {
+  await prisma.cartListItem.createMany({
+    data: ingredientIds.map((ingredientId) => ({
+      cartListId,
+      ingredientId,
+    })),
+  });
+};
+
+export const addCartList = async (recipeId: string, ingredientIds: number[]) => {
+  const supabaseServerClient = createServerActionClient<Database>({ cookies });
+
+  const {
+    data: { session },
+  } = await supabaseServerClient.auth.getSession();
+
+  if (!session) {
+    throw new Error("認証に失敗しました");
+  }
+
+  const cartList = await prisma.cartList.findMany({
+    include: {
+      CartListItem: true,
+    },
+    orderBy: {
+      displayOrder: "desc",
+    },
+  });
+
+  const foundCartList = cartList.find((c) => c.recipeId === recipeId);
+
+  if (foundCartList) {
+    // レシピが既にカート内に存在する場合は、カートに入っていない材料のみを追加する
+    const existingIngredientIds = foundCartList.CartListItem.map((item) => item.ingredientId);
+    const unAddedIngredientIds = ingredientIds.filter((id) => !existingIngredientIds.includes(id));
+    createCartListItem(foundCartList.id, unAddedIngredientIds);
```

Comment:
ここも別途メソッド切る必要ないと思います、

コメント（レシピが既にカート内に存在する場合は、カートに入っていない材料のみを追加する）で十分どのような処理をしたいか伝わると思うので！

```suggestion
    await prisma.cartListItem.createMany({
      data: unAddedIngredientIds.map((ingredientId) => ({
        ingredientId,
        cartListId: existingCartList.id,
      })),
    });
```

Reply:
@tuboihirokidesu 
ありがとうございます。恐らく私とツボイさんのコードを書く目的が異なっているように感じました。

何をやっているかを関数名で表現したいという意図ももちろんあります。しかし、関数化の重要な目的はメイン処理の差分を明確にすることで可読性を上げることにあります。

ツボイさんのコードはぱっと見でifとelseのコードの差分を見つけるのに時間がかかる印象です。

私のコードは共通的に行う処理がメソッド化されているので、引数によって差分を表現しています。

ですので、どちらのコードも個人的には問題ない認識です。ここら辺kumbikoさん交えて会話してみても面白そうです！